### PR TITLE
Stop clientless lobstrosities from fishing other lobstrosities and the necropolis chest

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
So, I've heard something funny happened on a downstream. A lobstrosity fished out another lobstrosity from the plasma river. And later another couple lobstrosities joined the fry, then again and again until it became a gang of over 50 lobstrosities that would absolutely terminate anyone that dared to interrupt their fishing party, as well as their FPS.

immagine

I've also blacklisted the single necropolis chest the lava fishing spot has, because it kinda sucks to get that chance to fish it up stolen by some random lobstrosity god-knows-where around the map.

## Why It's Good For The Game
Yes, this is very funny. However, getting your FPS butchered by dozens upon dozens of lobstrosities is not, and that's an honest issue.

## Changelog

:cl:
fix: Stop clientless lobstrosities from fishing other lobstrosities, which in turn can fish other lobstrosities and so on.
balance: Stop clientless lobstrosities from fishing the lavaland fishing spot chest.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
itd be cool if feeding them a bone key made them fish up necro chests. factory!

**Comment by Ghommie** [comment]
> itd be cool if feeding them a bone key made them fish up necro chests. factory!

If you code it.

**Comment by necromanceranne** [comment]
Lobstrocity family lavaloop cookout

**Comment by Jacquerel** [comment]
> itd be cool if feeding them a bone key made them fish up necro chests. factory!

Worth remembering that there's literally one fishable chest, once someone fishes it up there isn't one any more.

File: code/modules/fishing/sources/_fish_source.dm [hunk 1]
```diff
@@ -57,6 +57,11 @@ GLOBAL_LIST_INIT(specific_fish_icons, zebra_typecacheof(list(
 	var/explosive_malus = FALSE
 	/// If explosive_malus is true, this will be used to keep track of the turfs where an explosion happened for when we'll spawn the loot.
 	var/list/exploded_turfs
+	/// Mindless mobs that can fish will never pull up items on this list
+	var/static/list/profound_fisher_blacklist = typecacheof(list(
+		/mob/living/basic/mining/lobstrosity,
```
Comment (Jacquerel) [inline code/modules/fishing/sources/_fish_source.dm:62]: Didn't you add baby lobsters to the fishing list too, or do you not mind if they fish those up? Will it not lead to similar issues?

Comment (Ghommie) [reply code/modules/fishing/sources/_fish_source.dm:62]: It will lead to similar issues because baby lobstrosities also fish.

Comment (Jacquerel) [reply code/modules/fishing/sources/_fish_source.dm:62]: I'm dumb and forgot that this already includes them on account of being a subtype

**Pull request merged**